packages/frontend/core/src/desktop/dialogs/setting/account-setting/delete-account.tsx

```diff
@@ -0,0 +1,224 @@
+  const openModal = useCallback(() => {
+    setShowModal(true);
+  }, []);
+
+  return (
+    <SettingWrapper>
+      <SettingRow
+        name={
+          <span style={{ color: cssVarV2('status/error') }}>
+            {t['com.affine.setting.account.delete']()}
+          </span>
+        }
+        desc={t['com.affine.setting.account.delete.message']()}
+        style={{ cursor: 'pointer' }}
+        onClick={openModal}
+        data-testid="delete-account-button"
+      >
+        <ArrowRightSmallIcon />
+      </SettingRow>
+      {isTeamWorkspaceOwner ? (
+        <TeamOwnerWarningModal open={showModal} onOpenChange={setShowModal} />
+      ) : (
+        <DeleteAccountModal open={showModal} onOpenChange={setShowModal} />
+      )}
+    </SettingWrapper>
+  );
+};
+
+const TeamOwnerWarningModal = ({
+  open,
+  onOpenChange,
+}: {
+  open: boolean;
+  onOpenChange: (open: boolean) => void;
+}) => {
+  const t = useI18n();
+  const onConfirm = useCallback(() => {
+    onOpenChange(false);
+  }, [onOpenChange]);
+  return (
+    <ConfirmModal
+      open={open}
+      onOpenChange={onOpenChange}
+      title={t['com.affine.setting.account.delete.team-warning-title']()}
+      description={t[
+        'com.affine.setting.account.delete.team-warning-description'
+      ]()}
+      confirmText={t['Confirm']()}
+      confirmButtonOptions={{
+        variant: 'primary',
+      }}
+      onConfirm={onConfirm}
+      cancelButtonOptions={{
+        style: {
+          display: 'none',
+        },
+      }}
+    />
+  );
+};
+
+const DeleteAccountModal = ({
+  open,
+  onOpenChange,
+}: {
+  open: boolean;
+  onOpenChange: (open: boolean) => void;
+}) => {
+  const t = useI18n();
+  const {
+    workspacesService,
+    globalContextService,
+    serverService,
+    authService,
+  } = useServices({
+    WorkspacesService,
+    GlobalContextService,
+    ServerService,
+    AuthService,
+  });
+  const session = authService.session;
+  const account = useLiveData(session.account$);
+
+  const [email, setEmail] = useState('');
+  const { openPage, jumpToIndex } = useNavigateHelper();
+  const { openConfirmModal } = useConfirmModal();
+
+  const workspaces = useLiveData(workspacesService.list.workspaces$);
+  const currentWorkspaceFlavour = useLiveData(
+    globalContextService.globalContext.workspaceFlavour.$
+  );
+
+  const callback = useCallback(() => {
+    // if current account is deleted, switch to other workspace
+    if (currentWorkspaceFlavour === serverService.server.id) {
+      const localWorkspace = workspaces.find(
+        w => w.flavour !== serverService.server.id
+      );
+
+      if (localWorkspace) {
+        openPage(localWorkspace.id, 'all');
+      } else {
+        jumpToIndex();
+      }
+    }
+  }, [
+    currentWorkspaceFlavour,
+    jumpToIndex,
+    openPage,
+    serverService.server.id,
+    workspaces,
+  ]);
+
+  const handleDeleteAccount = useCallback(async () => {
```

Comment:
useAsyncCallback

Reply:
Do you mean to trigger the effect after deleting the account?
```
  const handleDeleteAccount = useCallback(async () => {
    try {
      const res = await authService.deleteAccount();
      if (res.success) {
        setDeleteSuccess(true);
      }
    } catch (err) {
      ...
    }
  }, [authService]);

  const callback = useCallback(() => {
    jumpToIndex();
  }, [jumpToIndex]);

  useEffect(() => {
    if (deleteSuccess) {
      openConfirmModal({
        ...
        onOpenChange: callback,
        onConfirm: callback,
        ...
      });
    }
  }, [callback, deleteSuccess, openConfirmModal]);
  ```

Reply:
move side effect to service